src/Utilities/TypeTraits/CMakeLists.txt

```diff
@@ -3,7 +3,7 @@
 
 spectre_target_headers(
```

Comment:
[100% completely optional, probably better in a separate PR, I can do it too] I was thinking of separating out `TypeTraits` into a separate library since they are used a lot more than pretty much everything else in `Utilities` combined. I was also thinking more generally of splitting `Utilities` into a bunch of small, but more specialized libs in the future since currently it has a lot of random utilities. It seems being a bit more organized/focused with individual libraries would make the code more manageable. Any thoughts on this idea?

Reply:
I agree - splitting off `TypeTraits` in particular and other facets in general does sound like a good idea. I'd say it's a bit orthogonal to this PR, though, so I'd favor revisiting this.

Reply:
Agreed :)